content/docs/atom-guide/dev-guide/materials/lua-material-functor-api.md

```diff
@@ -0,0 +1,316 @@
+* **GetShaderOptionDependencies()**: Returns a list of all shader options that may be set by the functor. Errors will be reported for options that are accessed without reporting the dependency. (Note that the `Material::SetSystemShaderOption` function in the C++ API may only be called on shader options that are *not* declared as dependencies or otherwised used by the material type).
+* **Process(context)**: This is the main function that runs when the material is *used for rendering*. It runs when the material is first initialized or whenever relevant material property values are changed. The `context` object provides access to the API for accessing the *material properties and shader(s)*.
+* **ProcessEditor(context)**: This is the main function that runs when the material is *being edited in tools*. It runs when the material is first initialized or whenever relevant material property values are changed. The `context` object provides access to the API for accessing *material properties and their metadata*.
+
+**Example**:
+```lua
+function GetMaterialPropertyDependencies()
+    return {"textureMap", "useTexture"}
+end
+ 
+function GetShaderOptionDependencies()
+    return {"o_metallic_useTexture"}
+end
+
+function Process(context)
+    local textureMap = context:GetMaterialPropertyValue_Image("textureMap")
+    local useTexture = context:GetMaterialPropertyValue_bool("useTexture")
+    context:SetShaderOptionValue_bool("o_metallic_useTexture", useTexture and textureMap ~= nil)
+end
+
+function ProcessEditor(context)
+    local textureMap = context:GetMaterialPropertyValue_Image("textureMap")
+    local useTexture = context:GetMaterialPropertyValue_bool("useTexture")
+
+    if(nil == textureMap) then
+        context:SetMaterialPropertyVisibility("useTexture", MaterialPropertyVisibility_Hidden)
+        context:SetMaterialPropertyVisibility("textureMapUv", MaterialPropertyVisibility_Hidden)
+        context:SetMaterialPropertyVisibility("factor", MaterialPropertyVisibility_Enabled)
+    elseif(not useTexture) then
+        context:SetMaterialPropertyVisibility("useTexture", MaterialPropertyVisibility_Enabled)
+        context:SetMaterialPropertyVisibility("textureMapUv", MaterialPropertyVisibility_Disabled)
+        context:SetMaterialPropertyVisibility("factor", MaterialPropertyVisibility_Enabled)
+    else
+        context:SetMaterialPropertyVisibility("useTexture", MaterialPropertyVisibility_Enabled)
+        context:SetMaterialPropertyVisibility("textureMapUv", MaterialPropertyVisibility_Enabled)
+        context:SetMaterialPropertyVisibility("factor", MaterialPropertyVisibility_Hidden)
+    end
+end
+```
+
+## Global Utility Functions
+
+These functions are available in the global scope.
+
+* **Error**(`string`): report an error message to AZ_Error.
+* **Warning**(`string`): report a warning message to AZ_Warning.
+* **Print**(`string`): report a message to AZ_TracePrintf.
```

Comment:
Should these functions be in section User Guide > Lua? Since they are in the global scope and apply to all Lua functions, not just Materials.

Reply:
They are in the global scope for lua material functor scripts. They are not available in the lua gameplay script context.

Reply:
Global scope just means they aren't inside some object like "context" (i.e. "context:GetMaterialPropertyValue")